src/LogOutputManager.spec.ts

```diff
@@ -156,4 +170,102 @@ describe('LogOutputManager ', () => {
+                { text: 'test1', levelFilter: 'INFO', includeFilter: 'INCLUDE', excludeFilter: null, expected: true },
+                { text: 'test1', levelFilter: 'INFO', includeFilter: null, excludeFilter: 'EXCLUDE', expected: true },
+                { text: 'test1', levelFilter: null, includeFilter: 'INCLUDE', excludeFilter: 'EXCLUDE', expected: true },
+                { text: 'test1', levelFilter: 'INFO', includeFilter: 'INCLUDE', excludeFilter: 'EXCLUDE', expected: true },
+            ];
+            itParam('lf ${value.levelFilter} if {$value.levelFilter} ' +
+              'ef ${value.excludeFilter}', mustIncludeParams, (params) => {
+                  const logLine = new LogLine(params.text, true);
+                  logOutputManager.setIncludeFilter(params.includeFilter);
+                  logOutputManager.setExcludeFilter(params.excludeFilter);
+                  logOutputManager.setLevelFilter(params.levelFilter);
+                  assert.equal(logOutputManager.matchesFilter(logLine), params.expected);
+              });
+        });
+        describe('non-mustinclude items true scenarios', () => {
+            let mustIncludeParams = [
+            { text: 'INFO test1 INCLUDE', levelFilter: null, includeFilter: null, excludeFilter: null, expected: true },
+            { text: 'INFO test1 INCLUDE', levelFilter: 'INFO', includeFilter: null, excludeFilter: null, expected: true },
+            { text: 'INFO test1 INCLUDE', levelFilter: null, includeFilter: 'INCLUDE', excludeFilter: null, expected: true },
+            { text: 'INFO test1 INCLUDE', levelFilter: null, includeFilter: null, excludeFilter: 'EXCLUDE', expected: true },
+            { text: 'INFO test1 INCLUDE', levelFilter: 'INFO', includeFilter: 'INCLUDE', excludeFilter: null, expected: true },
+            { text: 'INFO test1 INCLUDE', levelFilter: 'INFO', includeFilter: null, excludeFilter: 'EXCLUDE', expected: true },
+            { text: 'INFO test1 INCLUDE', levelFilter: null, includeFilter: 'INCLUDE', excludeFilter: 'EXCLUDE', expected: true },
+            { text: 'INFO test1 INCLUDE', levelFilter: 'INFO', includeFilter: 'INCLUDE', excludeFilter: 'EXCLUDE', expected: true },
+            ];
+            itParam('lf ${value.levelFilter} if {$value.levelFilter} ' +
+            'ef ${value.excludeFilter}', mustIncludeParams, (params) => {
+                const logLine = new LogLine(params.text, false);
+                logOutputManager.setIncludeFilter(params.includeFilter);
+                logOutputManager.setExcludeFilter(params.excludeFilter);
+                logOutputManager.setLevelFilter(params.levelFilter);
+                assert.equal(logOutputManager.matchesFilter(logLine), params.expected);
+            });
+        });
+        describe('non-must include items false scenarios', () => {
+            let mustIncludeParams = [
+          { text: 'INFO test1 INCLUDE', levelFilter: 'DEBUG', includeFilter: null, excludeFilter: null, expected: false },
```

Comment:
Ironically, the syntax of my mocha inspired unit testing framework, is _way_ more readable than the at-hand options I could find for mocha parameterized testing.. Wow.